build.gradle.kts

```diff
@@ -1,226 +1,193 @@
+    id("java") // Java support
     alias(libs.plugins.kotlin) // Kotlin support
-    alias(libs.plugins.gradleIntelliJPlugin) // Gradle IntelliJ Plugin
+    alias(libs.plugins.intelliJPlatform) // IntelliJ Platform Gradle Plugin
     alias(libs.plugins.changelog) // Gradle Changelog Plugin
-    alias(libs.plugins.testLogger) // Nice test logs
+    alias(libs.plugins.qodana) // Gradle Qodana Plugin
     alias(libs.plugins.kover) // Gradle Kover Plugin
-    jacoco // Code coverage
 }
 
-group = prop("pluginGroup")
-version = prop("pluginVersion")
+group = providers.gradleProperty("pluginGroup").get()
+version = providers.gradleProperty("pluginVersion").get()
 
-val lsp4jVersion = prop("lsp4jVersion")
-val flexmarkVersion = prop("flexmarkVersion")
+// Set the JVM language level used to build the project.
+kotlin {
+    jvmToolchain(17)
+}
 
 // Configure project's dependencies
 repositories {
-    mavenLocal()
     mavenCentral()
+
+    // IntelliJ Platform Gradle Plugin Repositories Extension - read more: https://plugins.jetbrains.com/docs/intellij/tools-intellij-platform-gradle-plugin-repositories-extension.html
+    intellijPlatform {
+        defaultRepositories()
+    }
+
+    mavenLocal()
     maven { url = uri("https://repository.jboss.org/nexus/content/repositories/snapshots") }
     maven { url = uri("https://repository.jboss.org/nexus/content/groups/public") }
     maven { url = uri("https://packages.jetbrains.team/maven/p/ij/intellij-dependencies") }
 }
 
-val lsp: Configuration by configurations.creating
+// Dependencies are managed with Gradle version catalog - read more: https://docs.gradle.org/current/userguide/platforms.html#sub:version-catalog
+dependencies {
+    testImplementation(libs.junit)
 
-sourceSets {
-    named("test") {
-        java.srcDir("src/test/java")
-        resources.srcDir("src/test/resources")
-    }
+    // IntelliJ Platform Gradle Plugin Dependencies Extension - read more: https://plugins.jetbrains.com/docs/intellij/tools-intellij-platform-gradle-plugin-dependencies-extension.html
+    intellijPlatform {
+        create(providers.gradleProperty("platformType"), providers.gradleProperty("platformVersion"))
+
+        // Plugin Dependencies. Uses `platformBundledPlugins` property from the gradle.properties file for bundled IntelliJ Platform plugins.
+        bundledPlugins(providers.gradleProperty("platformBundledPlugins").map { it.split(',') })
+
+        // Plugin Dependencies. Uses `platformPlugins` property from the gradle.properties file for plugin from JetBrains Marketplace.
+        plugins(providers.gradleProperty("platformPlugins").map { it.split(',') })
 
-    create("integrationTest") {
-        java.srcDir("src/it/java")
-        resources.srcDir("src/it/resources")
-        compileClasspath += sourceSets.main.get().output
-        compileClasspath += configurations.testImplementation.get()
-        runtimeClasspath += compileClasspath + sourceSets.test.get().output
+        instrumentationTools()
```

Comment:
when runnig locally a build, there is a warning about this usage : 
`w: file:///home/sbouchet/IdeaProjects/lsp4ij/build.gradle.kts:51:9: 'instrumentationTools(): Unit' is deprecated. Calling instrumentationTools() is no longer necessary.
`